File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):            
    answer = 0
    visited = [0] * n

    def bfs(a, visited, computers):
        visited[a] = 1
        queue = deque([a])
        
        while queue:
            x = queue.popleft()
            
            for i in range(n):
                if computers[x][i] == 1 and not visited[i]:
                    queue.append(i)
                    visited[i] = 1

    for i in range(n):
        if not visited[i]:
            bfs(i, visited, computers)
            answer += 1

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    var visited = Array(n).fill(0);
    
    function bfs(a, computers, visited) {
        var queue = [a];
        visited[a] = 1;

        while (queue.length > 0) {
            x = queue.pop();
            
            for (let i=0; i < n; i++) {
                if (visited[i] == 0 && computers[x][i] == 1) {
                    queue.push(i);
                    visited[i] = 1;
                }
            }
        }
    }
    
    for (let i=0; i < n; i++) {
        if (visited[i] == 0) {
            bfs(i, computers, visited);
            answer += 1;   
        }
    }
    
    return answer;
}